### ISSUE TITLE
Webhooks

### ISSUE THREAD

**Issue by ChachyDev** [dzikoysk/reposilite#844]
### Request details

Deployment notifications would send a webhook to the given URL in the configuration. For example, there could be Discord support which would display:
![image](https://user-images.githubusercontent.com/38067923/138142076-ecc7df2d-14a1-4e5a-a910-fea9f72f4229.png)

**Comment by hypherionmc** [comment]
You can use this library to handle the webhooks: https://github.com/MinnDevelopment/discord-webhooks

**Comment by ChachyDev** [comment]
Could be implemented as a separate plugin in reality now the API is present

**Comment by EthanDevelops** [comment]
Once plugins are fully implemented, it would be pretty easy to create plugins for different webhooks, as there are many API wrappers for services like Slack, Discord etc.

**Comment by TOTHT0MI** [comment]
After I finish #993, I'll be happy to implement it.
Although since webhooks are using a pretty simple format, I don't think a wrapper is really needed, it's just a plus dependency.

**Comment by joshuasing** [comment]
I have been working on this for the past few days, I will create a draft pull request as soon as I have gotten it all working smoothly.

**Comment by Saladoc** [comment]
This is something I want to make use of in the near future. @joshuasing are you still working on this? Are you in need of any help or feedback?

**Comment by joshuasing** [comment]
> This is something I want to make use of in the near future. @joshuasing are you still working on this? Are you in need of any help or feedback?

Yes, I've been slowly working on this however I've been really busy and haven't been able to get much done. Hopefully in a few weeks I will have a working prototype, however if you have the time and want to, you could start on this and I can help when I have time.

**Comment by IM-IgniteDEV** [comment]
Any update on that?

**Comment by joshuasing** [comment]
> Any update on that?

Sorry for the delay in my reply.  
Unfortunately I haven't made any more progress on this since I last commented, and I've somehow got even busier.

I would recommend that if anyone else wants to work on this, please do.  
I'm likely unable to work on this for at least the next 2 months.

**Comment by dzikoysk** [comment]
Don't worry, let me know if you'd like to work on it in the future :)

**Comment by TOTHT0MI** [comment]
Do you have any working code you feel like pushing, so we could build on top of it?

I may start working on this after some important events have completed next week, because this is a feature our org. could use, as we rely heavily on reposilite.

I don't guarantee it however...

**Comment by Koboo** [comment]
For all those who need webhook support **explicitly** for Discord, I wrote a plugin! :)

https://github.com/Koboo/reposilite-discord-webhook-plugin
